### DISCUSSION TITLE
Automatically Delete Fork Branch After PR Merged in Upstream Repository

### DISCUSSION THREAD

**Discussion by StarsbySea** [ScoopInstaller/Main#4696] [Q&A]
Hello everyone,

I'm looking for a way to set up my repository so that when my Pull Request gets merged into the upstream repository, the corresponding branch in my forked repository is automatically deleted. I'm wondering if there's a GitHub Action or any other method to achieve this behavior.

I've searched for existing solutions, but I haven't found any that specifically address this use case. If you have any suggestions or resources that might help, please share them with me. I'd really appreciate your input.

Thanks in advance!

**Comment by rashil2000** [comment]
I don't think this is possible, unless GitHub itself provides a way to automatically delete branches in the fork's Settings page.

You can set up a webhook server and listen to merge events from your fork, then call the GitHub API to delete the branches, but that's too much work IMO.
